app/javascript/pages/country/widget/widgets/widget-emissions/widget-emissions-selectors.js

```diff
@@ -0,0 +1,137 @@
+import { createSelector } from 'reselect';
+import isEmpty from 'lodash/isEmpty';
+import { format } from 'd3-format';
+import { getColorPalette } from 'utils/data';
+
+const EMISSIONS_KEYS = [
+  'Total including LUCF',
+  'Land-Use Change and Forestry',
+  'Agriculture'
+];
+
+// get list data
+const getData = state => state.data || null;
+const getLocationNames = state => state.locationNames || null;
+const getColors = state => state.colors || null;
+
+const getSortedData = createSelector([getData], data => {
+  if (!data || isEmpty(data)) return null;
+
+  const sortedData = {
+    data: [],
+    total: {}
+  };
+  Object.keys(data)
+    .filter(key => EMISSIONS_KEYS.includes(data[key].sector))
+    .forEach(key => {
+      if (data[key].sector === 'Total including LUCF') {
+        sortedData.total = data[key];
+      } else {
+        sortedData.data.push(data[key]);
+      }
+    });
+  return sortedData;
+});
+
+export const getChartData = createSelector([getSortedData], sortedData => {
+  if (!sortedData || !sortedData.data.length) return null;
+
+  const { data, total } = sortedData;
+  const chartData = [];
+  data[0].emissions.forEach((item, i) => {
+    chartData.push({
+      year: item.year,
+      e1_value: item.value,
```

Comment:
I dont like using kebab case when we can avoid it in js but it is ok for now.

Reply:
yeah me too... 😆  I don't know why I've used this nomenclature in this case